nsqd/protocol_v2.go

```diff
@@ -615,19 +615,20 @@ func (p *protocolV2) SUB(client *clientV2, params [][]byte) ([]byte, error) {
 	// last client can leave the channel between GetChannel() and AddClient().
 	// Avoid adding a client to an ephemeral channel / topic which has started exiting.
 	var channel *Channel
-	for {
+	for i := 1; ; i++ {
 		topic := p.nsqd.GetTopic(topicName)
 		channel = topic.GetChannel(channelName)
 		if err := channel.AddClient(client.ID, client); err != nil {
-			return nil, protocol.NewFatalClientErr(nil, "E_TOO_MANY_CHANNEL_CONSUMERS",
-				fmt.Sprintf("channel consumers for %s:%s exceeds limit of %d",
-					topicName, channelName, p.nsqd.getOpts().MaxChannelConsumers))
+			return nil, protocol.NewFatalClientErr(err, "E_SUB_FAILED", "SUB failed "+err.Error())
```

Comment:
hmm, now this is where the exiting will be checked, and we won't retry for ephemeral channels exiting, as intended

Reply:
I think that's fine, if we looked up a channel that is indeed exiting we _want_ this to fail here.